File: TaskList.Domain/Entities/Tasks/UserTask.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TaskList.Domain.Entities.Tasks;

public class UserTask
{
    [Key]
    public int Id { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public required DateTime CreatedDateTime { get; set; }
    public DateTime ConslusionDateTime { get; set; }
    public bool Done { get; set; } = false;

    public static UserTask Create(string title, string description)
        => new()
        {
            Title = title,
            Description = description,
            CreatedDateTime = DateTime.Now,
        };
}

File: src/frontend/src/types/Tasks/UserTask.ts
export type UserTask = {
    id?: number
    title: string
    description: string
    createdDateTime: Date
    releaseDateTime: Date
    conslusionDateTime?: Date
    done: boolean
}

export type CreateUserTask = Pick<UserTask, 'title' |  'description'> & {releaseDateTime: string}
